app/apollo/resolvers/cluster.js

```diff
@@ -111,9 +110,13 @@ const clusterResolvers = {
           throw new NotFoundError(context.req.t('Could not find the cluster with Id {{clusterId}}.', {'clusterId':clusterId}), context);
         }
 
-        await validAuth(me, org_id, ACTIONS.READ, TYPES.CLUSTER, queryName, context, [clusterId, cluster.name]);
+        logger.info({req_id, user, org_id, clusterId}, `${queryName} found matching cluster`);
+
+        logger.info({req_id, user, org_id, clusterId}, `${queryName} validating`);
+
+        await validAuth(me, org_id, ACTIONS.READ, TYPES.CLUSTER, queryName, context, [clusterId, cluster.registration.name]);
```

Comment:
fallback to cluster.name, e.g. `cluster.registration.name || cluster.name`

Reply:
fixed